File: smart-campus-client/src/pages/module-c-maintenance-incident-ticketing/types/ticket.ts
export type Priority = "LOW" | "MEDIUM" | "HIGH";
export type TicketStatus = "OPEN" | "IN_PROGRESS" | "RESOLVED" | "CLOSED" | "REJECTED";

export interface Comment {
  id: string;
  authorId: string;
  authorName: string;
  authorRole: "USER" | "TECHNICIAN" | "ADMIN";
  content: string;
  createdAt: string;
  updatedAt?: string;
}

export interface Ticket {
  id: string;
  backendId?: number;
  title: string;
  category: string;
  description: string;
  priority: Priority;
  status: TicketStatus;
  resourceLocation: string;
  images: string[];
  assignedToName?: string;
  resolutionNote?: string;
  rejectionReason?: string;
  comments: Comment[];
  createdAt: string;
  updatedAt: string;
}

File: smart-campus-api/src/main/java/com/smartcampus/ticket/model/Ticket.java
package com.smartcampus.ticket.model;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import com.smartcampus.ticket.enums.Priority;
import com.smartcampus.ticket.enums.Status;
import com.smartcampus.user.entity.User;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "tickets")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Ticket {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    private String category; // hardware, software, etc.

    @Column(columnDefinition = "TEXT")
    private String description;

    private String resourceLocation;


    @Enumerated(EnumType.STRING)
    private Priority priority; // LOW, MEDIUM, HIGH

    @Enumerated(EnumType.STRING)
    private Status status; // OPEN, IN_PROGRESS, RESOLVED, CLOSED, REJECTED

    @Column(columnDefinition = "TEXT")
    private String resolutionNote;

    @Column(columnDefinition = "TEXT")
    private String rejectionReason;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User createdBy;

    @ManyToOne
    @JoinColumn(name = "technician_id")
    private User technician;

    @OneToMany(mappedBy = "ticket", cascade = CascadeType.ALL)
    private List<TicketAttachment> attachments = new ArrayList<>();

    @OneToMany(mappedBy = "ticket", cascade = CascadeType.ALL)
    private List<TicketComment> comments = new ArrayList<>();

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
